app/src/main/java/com/github/se/polyfit/ui/screen/PostInfoScreen.kt

```diff
@@ -61,31 +60,32 @@ fun PostInfoScreenContent(
     padding: PaddingValues = PaddingValues(),
     viewPostViewModel: ViewPostViewModel = hiltViewModel(),
 ) {
-  val posts = viewPostViewModel.getAllPost().observeAsState(initial = listOf())
-  val isFetching by viewPostViewModel.isFetching.collectAsState()
 
   Scaffold(modifier = Modifier.padding(padding)) {
-    if (isFetching) {
+    val isFetching = viewPostViewModel.isFetching.value
```

Comment:
This needs to be collected and observed as a state in order for the UI to be updated dynamically.